### ISSUE TITLE
Google App Engine deployment hangs

### ISSUE THREAD

**Issue by cheekychops** [Shippable/support#999]
I have a Java GAE app which I want to deploy using maven. The deployment hangs with 

Please open the following URL in your browser:
  https://accounts.google.com/o/oauth2/auth?access_type=offline&approval_prompt=force&client_id=550516889912.apps.googleusercontent.com&redirect_uri=urn:ietf:wg:oauth:2.0:oob&response_type=code&scope=https://www.googleapis.com/auth/appengine.admin%20https://www.googleapis.com/auth/cloud-platform

**Comment by manishas** [comment]
Tagging @rageshkrishna to take a look

**Comment by rageshkrishna** [comment]
It looks like you may have configured your GAE deployment to use OAuth for authentication. I'm not sure if there's a way to get this to work because you cannot interactively enter the authentication key to proceed with the OAuth flow. 

You'll need to use username/password to authenticate instead. Here's a snippet from one of our sample projects that does this:

````
  - echo "$GAE_PASSWORD" | $GAE_DIR/google_appengine/appcfg.py -e "$EMAIL" --passin update .
````
The complete sample is available at https://github.com/shippableSamples/sample-python-datastore-appengine/blob/master/shippable.yml. It's a python project, but I think you should be able to do the same thing from any project because all it's doing is invoking the GAE SDK and providing a username and password.

**Comment by manishas** [comment]
Closing this. Please re-open if you're still running into this

**Comment by aaronwest** [comment]
NOTE: The `--passin` flag is no longer available in appcfg.py.  Not sure when it was removed but it isn't available in GAE SDK v1.9.24.
